services/src/handlers/datasets.rs

```diff
@@ -0,0 +1,143 @@
+
+    #[tokio::test]
+    async fn test_list_datasets() -> Result<()> {
+        // TODO: use new tests helpers once they are merged
+        let ctx = InMemoryContext::default();
+
+        ctx.user_db()
+            .write()
+            .await
+            .register(
+                UserRegistration {
+                    email: "foo@bar.de".to_string(),
+                    password: "secret123".to_string(),
+                    real_name: "Foo Bar".to_string(),
+                }
+                .validated()
+                .unwrap(),
+            )
+            .await
+            .unwrap();
+
+        let session = ctx
+            .user_db()
+            .write()
+            .await
+            .login(UserCredentials {
+                email: "foo@bar.de".to_string(),
+                password: "secret123".to_string(),
+            })
+            .await
+            .unwrap();
+
+        let descriptor = VectorResultDescriptor {
+            data_type: VectorDataType::Data,
+            spatial_reference: SpatialReferenceOption::Unreferenced,
+            columns: Default::default(),
+        };
+
+        let ds = AddDataSet {
+            name: "OgrDataSet".to_string(),
+            description: "My Ogr data set".to_string(),
+            source_operator: "OgrSource".to_string(),
+            result_descriptor: descriptor.clone().into(),
+        };
+
+        let meta = StaticMetaData {
+            loading_info: OgrSourceDataset {
+                file_name: Default::default(),
+                layer_name: "".to_string(),
+                data_type: None,
+                time: Default::default(),
+                columns: None,
+                default_geometry: None,
+                force_ogr_time_filter: false,
+                on_error: OgrSourceErrorSpec::Skip,
+                provenance: None,
+            },
+            result_descriptor: descriptor,
+        };
+
+        let _id = ctx
+            .data_set_db_ref_mut()
+            .await
+            .add_data_set(UserId::new(), ds.validated()?, Box::new(meta))
+            .await?;
+
+        let res = warp::test::request()
+            .method("GET")
+            .path(&format!(
+                "/datasets?{}",
+                &serde_urlencoded::to_string([
+                    ("order", "NameDesc"),
+                    ("offset", "0"),
+                    ("limit", "2"),
+                ])
+                .unwrap()
+            ))
+            .header("Content-Length", "0")
+            .header(
+                "Authorization",
+                format!("Bearer {}", session.id.to_string()),
+            )
+            .reply(&list_datasets_handler(ctx))
+            .await;
+
+        assert_eq!(res.status(), 200);
+
+        let body: String = String::from_utf8(res.body().to_vec()).unwrap();
+        let result = serde_json::from_str::<Vec<DataSetListing>>(&body);
+        assert!(result.is_ok());
+        assert_eq!(result.unwrap().len(), 1);
```

Comment:
Maybe just check against JSON output, then we know that the result is correct and also test the output schema.

Reply:
a873364b17e4d25c8a4aecc3a5f2a1b7b2dad635